modules/grpc/otel/otel-protobuf-parser.hpp

```diff
@@ -26,6 +26,7 @@
 #include "otel-protobuf-parser.h"
 
 #include "compat/cpp-start.h"
+#include "syslog-ng.h"
```

Comment:
just curious, all the other `.hpp` headers include `syslog.h` outside the compact block, why the extern "C" is needed in this case?

Reply:
I will check this. The cpp-helper says this:

'syslog-ng.h needs to come first even in cpp/hpp files.  You don't need cpp-start/end wrapper around syslog-ng.h.`

But I am not sure this is correct, because syslog-ng.h pulls glib and glib does some macro magic completely differently in case of C++ and C.

Reply:
I remembered completely the opposite of the truth.
We need syslog-ng.h outside of the external C block especially because of the glib behavior.

Thanks for your comment. I will drop the last patch.

Reply:
also, i can see this in the `syslog-ng.h`

```
 * Header ordering:
 *   - syslog-ng.h needs to come first even in cpp/hpp files.  You don't
 *     need cpp-start/end wrapper around syslog-ng.h.  This include can
 *     happen indirectly, e.g.  if you include any other header that should
 *     take care of syslog-ng.h
```